client/spa/js/learning-resource/learning-resource.controller.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+var Backbone = require('../vendor/index').Backbone;
+var $ = require('../vendor/index').$;
+var Model = require('./learning-resource.model');
+var View = require('./learning-resource.view');
+
+module.exports = Backbone.Controller.extend({
+
+  routes: {
+    'learning-resource/:': 'list',
+    'learning-resource/:id': 'showLearningResource'
+  },
+
+  initialize: function(){
+    this.options.container = this.options.container || 'body';
+    this.model = new Model();
+    this.view = new View({model: this.model});
```

Comment:
I've been trying to move the `this.model` to fetchModel, the front-end it seems to be working fine, but for the jasmine test I get this error:

<code>Chrome 40.0.2214 (Windows 7) Learning resource controller when calling showLearningResource with a valid learning resource id, fetches the model FAILED
        Error: spyOn could not find an object to spy upon for fetch()
            at Object.<anonymous> (C:/Users/JSTADR~1/AppData/Local/Temp/25ac7c575ff07e6c139cf69ca01edbec55aab123.browserify:300:23 <- client\spa\js\learning-resource\spec\learning-resource.controller.spec.js:59:0)
...................
Chrome 40.0.2214 (Windows 7): Executed 24 of 26 (1 FAILED) (skipped 2) (1.855 secs / 0.104 secs)</code>

Here's the part in the controller.spec:
```javascript
it('with a valid learning resource id, fetches the model', function(){
      spyOn(controller.model, 'fetch').and.callFake(success);
      var cb = function(err, view){
        expect(err).toBeNull();
        expect(controller.model.get('title')).toEqual('JavaScript Is Sexy');
      };
      controller.showLearningResource(1, cb);
    });
```

and the controller.js:
```javascript
  showLearningResource: function(learningResourceId, cb){
    this.fetchModel(learningResourceId, function(err, model){
      var view;

      if (err){
        this.model = new Model();
        view = this.renderError();
      }
      else {
        if (this.view) this.view.remove();
        this.view = new View({model: model});
        view = this.renderView();
      }

      if (cb){
        cb(err, view);
      }

    }.bind(this));
  },

  fetchModel: function(learningResourceId, cb){
    this.model = new Model(); //initialize model to allow property setting
    this.model.set({id: learningResourceId});

    this.model.fetch({
      success: function(model, response, options){
        //console.log(model);
        cb(null, model);
      },
      error: function(model, response, options){
        //console.error(response);
        cb(response, model);
      }
    });
  },
```
Looks like jasmine can't see the model anymore? If the `this.model` was left in initialization, it would pass. I'm still not sure what the difference is between initializing it and putting it where you suggested.